### PR TITLE
Upgrade 7zip in ArchiveFilesV2 task

### PR DESCRIPTION
**Pull request by ismayilov-ismayil**
**Task name**: ArchiveFilesV2 

**Description**: 7zip v19 has vulnerability, upgraded to v23 

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N)  N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

Related WI: [AB#2120588](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2120588)

### CODE REVIEW

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

File: _generated/ArchiveFilesV2/make.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "archivePackages": [
             {
                 "archiveName": "7zip.zip",
-                "url": "https://vstsagenttools.blob.core.windows.net/tools/7zip/4/7zip.zip",
+                "url": "https://vstsagenttools.blob.core.windows.net/tools/7zip/5/7zip.zip",
```
Comment (DergachevE) [inline _generated/ArchiveFilesV2/make.json:6]: Do I understand correctly that you are upgrading the major version of `7zip`? Are there any breaking changes? How would it be tested to detect any possible regression?

Comment (ismayilov-ismayil) [reply _generated/ArchiveFilesV2/make.json:6]: Yes, it's major version upgrading.
It is tested by different archive file formats on windows.

**Review by DergachevE — APPROVED** [review]

**Comment by mattdean98** [comment]
I would like to inform you that we have discovered an issue with the updated version in the compression process, while the extraction flow remains the same. Our analysis shows that the errors reported are linked to the recent changes made in this specific pull request. If a pipeline involves both compression and extraction processes, it may experience failures due to using different versions of 7zip simultaneously (v23.01 for compression and v19.00 for extraction). Thank you.

**Comment by HejdasGonnaHate** [comment]
> I would like to inform you that we have discovered an issue with the updated version in the compression process, while the extraction flow remains the same. Our analysis shows that the errors reported are linked to the recent changes made in this specific pull request. If a pipeline involves both compression and extraction processes, it may experience failures due to using different versions of 7zip simultaneously (v23.01 for compression and v19.00 for extraction). Thank you.

Same issue for me. We had to revert back to ArchiveFilesV1 task which uses 7zip version 16.00

**Comment by synt-acks** [comment]
> > I would like to inform you that we have discovered an issue with the updated version in the compression process, while the extraction flow remains the same. Our analysis shows that the errors reported are linked to the recent changes made in this specific pull request. If a pipeline involves both compression and extraction processes, it may experience failures due to using different versions of 7zip simultaneously (v23.01 for compression and v19.00 for extraction). Thank you.
> 
> Same issue for me. We had to revert back to ArchiveFilesV1 task which uses 7zip version 16.00

We ultimately opted for using ".zip" files, which effectively solved our issue without the need to revert to a previous version of ArchiveFiles.

**Comment by wilbit** [comment]
I believe you have a high-urgency bug on your hands. 
It seems you've updated ArchiveFiles task but didn't update ExtractFiles task. 
This broke some stages of our Release pipelines - hope you can release a fix soon!

**Pull request merged**